File: rb/55.rb
def palindromic? n
  (s = n.to_s) == s.reverse
end

def lychrel? n
  50.times do
    n += n.to_s.reverse.to_i
    return false if palindromic? n
  end

  true
end

p (1..10000).select { |n| lychrel? n }.size

File: hs/55.hs
palindromic n = s == reverse s where s = show n

revsum n = n + read (reverse $ show n) :: Integer

lychrel' i n | i == 50 = True
             | palindromic n && i > 0 = False
             | otherwise = lychrel' (i + 1) (revsum n)

lychrel = lychrel' 0

main = print $ length $ filter lychrel [1..9999]
